src/importexport/mei/internal/meiimporter.cpp

```diff
@@ -134,12 +135,30 @@ bool MeiImporter::read(const muse::io::path_t& path)
 
     success = success && this->readMeiHead(root);
 
-    success = success && this->readScore(root);
-
     pugi::xml_attribute xmlId = root.attribute("xml:id");
+    bool hasRootXmlId = false;
     if (xmlId && !String(xmlId.value()).empty()) {
-        m_score->setMetaTag(u"xml:id", String(xmlId.value()));
-        // Do not keep a xml:id map when having a xml:id seed.
+        hasRootXmlId = true;
+        String xmlIdStr = String(xmlId.value());
+        if (xmlIdStr.startsWith(u"mscore-")) {
+            // Keep a global flag since we are going to read them only if mei@xml:id is given with LastEID
```

Comment:
LastEID is not a thing anymore now that MuseScore uses randomly generated IDs. So perhaps this comment should be adjusted.